src/frequenz/client/dispatch/__main__.py

```diff
@@ -17,7 +17,20 @@
 from prompt_toolkit.patch_stdout import patch_stdout
 from prompt_toolkit.shortcuts import CompleteStyle
 
-from ._cli_types import FuzzyDateTime, FuzzyIntRange, FuzzyTimeDelta, SelectorParamType
+from frequenz.client.dispatch.types import (
+    EndCriteria,
+    Frequency,
+    RecurrenceRule,
+    Weekday,
+)
+
+from ._cli_types import (
+    FuzzyDateTime,
+    FuzzyIntRange,
+    FuzzyTimeDelta,
+    JsonParamType,
+    SelectorParamType,
+)
```

Comment:
At this point the `cli` subpackage would only contain one file (if we keep __main__ were it is).
I have no problem changing the location(s) though

Reply:
I see, so you use `python -m frequenz.client.dispatch`, right? Yeah, I think I would make it `frequenz.client.dispatch.cli` to keep stuff separated, I don't think it is a good idea to pollute the top-level package used from other downstream projects with the CLI. Even more, eventually I would like to ship it as a separate `pip` package. So you could have a `src/frequenz/client/dispatch/cli/__init__.py`, `src/frequenz/client/dispatch/cli/_main.py` and `src/frequenz/client/dispatch/cli/_types.py` for example.

But for a separate PR:
* #49

Reply:
Yes. You need to use multiple `pyproject.toml` files, it would be like a tiny mono-repo, but I guess since both would be so tied together it is fine. Another option would be to make it a separate repo but I guess that will be more overkill.

It will require some structural changes, like they should probably live in different parts of `src/`, like `src/frequenz/client/dispatch` and `src/frequenz/cli/dispatch`, and might need some adjustments to repo-config, so definitely not in the short term, but in some future we might go for it.